src/server/game/Entities/Player/Player.cpp

```diff
@@ -15840,6 +15840,8 @@ void Player::RewardQuest(Quest const* quest, uint32 reward, Object* questGiver,
         {
             if (quest->RequiredItemCount[i] > 0 && itemTemplate->Bonding == BIND_QUEST_ITEM && !quest->IsRepeatable() && !HasQuestForItem(quest->RequiredItemId[i], quest_id, true))
                 DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
+            else
+                DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
```

Comment:
then why not removing the `if` condition?

Reply:
Ok I’ll try that and test it